lib/dialects/abstract/connection-manager.js

```diff
@@ -295,6 +295,11 @@ class ConnectionManager {
   releaseConnection(connection) {
     return this.pool.release(connection).tap(() => {
       debug('connection released');
+    }).catch((e) => {
+      // ISSUE(7625) No-op if generic-pool is unable to release the connection, it may have already been released.
+      if (!/Resource not currently part of this pool/.exec(e.message)) {
```

Comment:
@jaredjenkins After further examination all you need to check against is the borrowed getter which is the same as `this.pool._resourceLoans.size`.  You don't need to check if the connection exists in the pool's `_resourceLoans` property.

```javascript
if (err.code === 'PROTOCOL_CONNECTION_LOST' && this.pool.borrowed > 0) {
```